source/common/network/utility.cc

```diff
@@ -208,6 +208,45 @@ Address::InstanceConstSharedPtr Utility::getLocalAddress(const Address::IpVersio
   return ret;
 }
 
+bool Utility::isLocalConnection(const Network::ConnectionSocket& socket) {
+  const auto& remote_address = socket.remoteAddress();
+  if (remote_address->type() == Envoy::Network::Address::Type::Pipe ||
```

Comment:
@PiotrSikora would you agree with that proposal? I can do one more patch on top.

Reply:
I'm not sure this is _always_ true (e.g. imagine transparent proxy where outgoing connection is bound to an IP address that's not local and it's the same as the destination it tries to connect to, i.e. outgoing IP == destination IP).

I won't go as far as objecting this change, since it's clearly an useful optimization that's valid in majority of the cases, but please add a comment explaining why it's there.

cc @htuch

Reply:
From what I know the non local bind is used only when opening a connection (client side) while this feature is when accepting a connection (server side). If the connection was transparently proxied with a non local address which happens to be the address envoy is listening on then even without this check the connection would be classified as local. 

If the transparent proxy is correctly working then the non local address would be the actual address of the client. If the address is the same as the envoy listening address then the client and envoy are on the same host so there is no problem.

Reply:
Sorry, my explanation wasn't clear. What I meant is client's connection bound to a non-local IP (connecting to the same IP) being intercepted by `iptables` and redirected to Envoy, which then sees same IP for client and destination.

This is rather unusual flow, so I'm fine with having this optimization in the code, but I'd like to have comment explaining why "remote address (connecting IP) == local address (listening IP)" is considered a local connection.